xapian-core/weight/boolweight.cc

```diff
@@ -86,4 +92,14 @@ BoolWeight::get_maxextra() const
     return 0;
 }
 
+BoolWeight *
+BoolWeight::create_from_parameters(const char * p) const
+{
+    if (*p == '\0')
+	return new Xapian::BoolWeight();
+    if (*p == ' ')
+	throw InvalidArgumentError("No parameters are required for BoolWeight");
+    return new Xapian::BoolWeight();
```

Comment:
This is needlessly complicated, and yet not really correct.  Your code does:

* Empty string -> OK
* Non-empty string with space as first character -> bad
* Non-empty string with character other than space as first character -> OK

There are really just two cases:

* Empty string -> OK
* Non-empty string -> bad